spec/services/idv/steps/in_person/address_step_spec.rb

```diff
@@ -76,19 +76,14 @@
 
       context 'with secondary capture enabled' do
         let(:capture_secondary_id_enabled) { true }
-        it 'excludes same_address_as_id from session' do
-          expect(flow.flow_session[:pii_from_user]).not_to include(
-            same_address_as_id:,
-          )
-        end
-
-        it 'sets other values in flow session' do
+        it 'sets the values in flow session' do
           expect(flow.flow_session[:pii_from_user]).to include(
             address1:,
             address2:,
             city:,
             zipcode:,
             state:,
+            same_address_as_id:,
```

Comment:
Let's expand this test to ensure that both address create and update are tested for this step.

Reply:
In particular, I want to ensure the following conditional in `address_step.rb` is fully exercised by the test.

```ruby
if capture_secondary_id_enabled? && updating_address? && attr == :same_address_as_id
  flow_session[:pii_from_user][:same_address_as_id] = 'false'
else
  flow_session[:pii_from_user][attr] = flow_params[attr]
end
```

Reply:
Thank you for this suggestion. I've pushed up the tests looking at the different conditionals. Please take a look when you have time.